platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrClientFactory.java

```diff
@@ -30,5 +30,16 @@
    * @return the newly created {@code SolrClient}
    * @throws IllegalArgumentException if <code>core</code> is <code>null</code>
    */
-  public SolrClient newClient(String core);
+  SolrClient newClient(String core);
+
+  SolrClient newClient(String collection, String alias, SolrCollectionConfiguration configuration);
+
+  /**
+   * Returns whether or not the provider connects to SolrCloud instance or Standalone. Solr Cloud
+   * provides different capabilities that upstream Index and Storage providers need knowledged of
+   * (e.g. Collection Aliases in SolrCloud).
+   *
+   * @return Provider is a SolrCloud provider.
+   */
+  boolean isSolrCloud();
```

Comment:
:question: Non-SolrCloud factories have been deprecated.  Do we still need this?

Reply:
probably not. since `httpSolrClientFactory` hasn't been removed. the though is make sure it is still compatible in non-solrcloud mode.

Reply:
I wonder if we should just have code check the system property directly to see if it is in SolrCloud mode instead of adding it to this interface.  Then we can remove those checks once we remove the HTTP client factory.